File: JAVASCRIPT/mock/mock2/revString.js
var x = "abc";
function stringReversed(x) {
    for(let i = 0; i < x.length; i++) {
        let temp = x[i];
        x[i] = x[x.length - 1 -i];
        x[x.length - 1 -i] = temp;

    }
    console.log(x);
}
//stringReversed(x);
console.log(x.reverse());

File: JAVA/mock/DSA2(Easy)/RevString.java
import java.util.Scanner;

public class RevString {
    private static void stringReversed(String str) {
        int len = str.length();
        char[] ch = new char[len];
        for(int i = 0; i < len; i++) {
            ch[i] = str.charAt(i);
        }
        for(int i = 0; i < len / 2 ; i++) {
            char temp = ch[i];
            ch[i] = ch[len - 1 - i];
            ch[len - 1 - i] = temp;
        }
        str = String.valueOf(ch);
        System.out.println(str);
    }
   
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        stringReversed(str);
        sc.close();
    }
}
